polkadot/xcm/xcm-executor/src/lib.rs

```diff
@@ -825,7 +826,29 @@ impl<Config: config::Config> XcmExecutor<Config> {
 				let old_holding = self.holding.clone();
 				let result = Config::TransactionalProcessor::process(|| {
 					let deposited = self.holding.saturating_take(assets);
+					let mut all_failed = true;
+					let mut failed_deposits = Vec::new();
 					for asset in deposited.into_assets_iter() {
+						let asset_result = Config::AssetTransactor::deposit_asset(
+							&asset,
+							&beneficiary,
+							Some(&self.context),
+						);
+						match asset_result {
+							Ok(_) => {
+								all_failed = false;
+							}
+							Err(_err) => {
```

Comment:
We need to explicitly match against the error happening because of ED not satisfied. We only want to retry on that specific error.

Reply:
I've decided to go with retry for any error.